meson.build

```diff
@@ -385,6 +412,19 @@ if get_option('enable-xkbregistry')
         libxkbregistry_link_deps += libxkbregistry_def
         libxkbregistry_link_args += '/DEF:' + libxkbregistry_def.full_path()
     endif
+    if xkbcommon_project_version[0] != '1'
+        # The versioning used for the shared libraries assumes that the major
+        # version of xkbcommon as a whole will increase to 2 if and only if there
+        # is an ABI break, at which point we should probably bump the SONAME of
+        # all libraries to .so.2.
+        error('We probably need to bump the SONAME of libxkbregistry')
+    else
+        # To avoid an unnecessary SONAME bump, xkbcommon 1.x.y produces
+        # libxkbregistry.so.0.x.y.
+        libxkbregistry_version = '.'.join(
+            ['0', xkbcommon_project_version[1], xkbcommon_project_version[2]]
+        )
+    endif
```

Comment:
how about moving this somewhere near the top and set a more generic `soname_version` variable that is then used for all three .so files. I don't se a future where we need to have different sonames for the three libraries, so this could be done without duplicating.

Reply:
Done